bofire/strategies/shortest_path.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional, Tuple
+
+import cvxpy as cp
+import numpy as np
+import pandas as pd
+
+from bofire.data_models.constraints.api import (
+    LinearConstraint,
+    LinearEqualityConstraint,
+    LinearInequalityConstraint,
+)
+from bofire.data_models.domain.api import Constraints, Inputs
+from bofire.data_models.features.api import ContinuousInput
+from bofire.data_models.strategies.api import ShortestPathStrategy as DataModel
+from bofire.strategies.strategy import Strategy
+
+
+class ShortestPathStrategy(Strategy):
+    def __init__(
+        self,
+        data_model: DataModel,
+        **kwargs,
+    ):
+        self.start = pd.Series(data_model.start)
+        self.end = pd.Series(data_model.end)
+        self.atol = data_model.atol
+        super().__init__(data_model=data_model, **kwargs)
+
+    @property
+    def continuous_inputs(self) -> Inputs:
+        return self.domain.inputs.get(ContinuousInput)  # type: ignore
+
+    def get_linear_constraints(
+        self, constraints: Constraints
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        inputs = self.continuous_inputs
+        keys = inputs.get_keys()
+        b = np.array([c.rhs for c in constraints])  # type: ignore
+        A = np.zeros([len(constraints), len(inputs)])
+        for i, c in enumerate(constraints):
+            assert isinstance(c, LinearConstraint)
+            for key, coef in zip(c.features, c.coefficients):
+                feat = inputs.get_by_key(key)
+                assert isinstance(feat, ContinuousInput)
+                if feat.is_fixed():
+                    b[i] -= feat.fixed_value()[0] * coef  # type: ignore
+                else:
+                    A[i, keys.index(key)] = coef
+        return A, b
+
+    def step(self, start: pd.Series) -> pd.Series:
+        inputs = self.continuous_inputs
+        lower, upper = inputs.get_bounds(
```

Comment:
Not sure, but would it make sense to get rid of the equality constraint by using the reduced domain? Then you would need compute the intersection between the boundary of the box and the connection between start and end which does not involve an optimization problem?

Reply:
Hmm, interesting idea. But what is with linear inequalities? We also have to account for them, or? Or do inequalities do not cause a problem? I am not sure. And formulating it as an optimization problem would give the possibility for more complex local constraints which are not just rectangular. What do you think? Keep it or reformulating it?

Reply:
I thought more about it, and want to keep it as optimization problem, as it is also not straight forward with the reduced domain.